bitcoin/src/sign_message.rs

```diff
@@ -128,8 +125,9 @@ mod message_signing {
         /// Create a `MessageSignature` from a byte slice.
         #[deprecated(since = "TBD", note = "Use `from_byte_array` instead.")]
         pub fn from_slice(bytes: &[u8]) -> Result<MessageSignature, MessageSignatureError> {
-            let byte_array: [u8; 65] =
-                bytes.try_into().map_err(|_| MessageSignatureError::InvalidLength)?;
+            let byte_array: [u8; 65] = bytes.try_into().map_err(|_| {
+                MessageSignatureError::InvalidEncoding(secp256k1::Error::InvalidMessage)
```

Comment:
This is just sweeping the problem under the rug (but admittedly the `secp256k1::Error` is already too bad, though there was a PR recently to make signature "parsing" infallible) and even if it was the best approach `InvalidSignature` would be more appropriate. We literally need something like `enum FromSliceError { InvalidLength, InvalidData(MessageSignatureError) }`. Though looking at `MessageSignatureError` that one is also quite messed up and I think long-term this function will become infallible.

But also, thinking about it more, to be in-line with deprecation we should maybe keep `MessageSignatureError` as-is and create a set of new, focused errors that can be converted to it. Then deprecate it.